src/agent/dfs_tree_search_with_stack.py

```diff
@@ -110,6 +118,9 @@ def update_new_node(self, tree: ProofQTree, state: ProofState, action: ProofActi
         elif self.language == ProofAction.Language.LEAN:
             description_match = DynamicLeanProofExecutor.ProofFinishedDescription
             qed_tac = ["end"]
+        elif self.language == ProofAction.Language.ISABELLE:
+            description_match = DynamicIsabelleProofExecutor.ProofFinishedDescription
+            qed_tac = ["qed"]
```

Comment:
This will only be used when a proof is finished but not closed -- therefore, 'qed' is correct